src/database/firebase.tsx

```diff
@@ -18,6 +19,119 @@ const config = {
   appId: APP_ID,
 };
 
-firebase.initializeApp(config);
+!firebase.apps.length ? firebase.initializeApp(config) : firebase.app();
 
+const database = firebase.firestore();
+const treeCollection = database.collection('trees');
+const commentCollection = database.collection('comments');
+const additionalCollection = database.collection('additional');
+
+/*
+ * checkID validates that this ID exists in the `trees` table.
+ */
+export const checkID = async (id: string): Promise<boolean> => {
+  try {
+    const doc = await treeCollection.doc(id).get();
+    return doc.exists;
+  } catch (e) {
+    console.warn(e);
+    throw e;
+    // TODO: Add error handling.
+  }
+};
+
+/*
+ * getTree queries the `trees` table and returns a Tree if the ID is found and an empty entry otherwise.
```

Comment:
gonna be nitpicky pls address the linter error on the char count!

Reply:
also for nice comment formatting, we should do 
```
/**
*
*/
```
with double star on first line